tutorials-v4/time-evolution/012_floquet_formalism.md

```diff
@@ -0,0 +1,276 @@
+---
+jupyter:
+  jupytext:
+    text_representation:
+      extension: .md
+      format_name: markdown
+      format_version: '1.3'
+      jupytext_version: 1.13.8
+  kernelspec:
+    display_name: Python 3 (ipykernel)
+    language: python
+    name: python3
+---
+
+# Floquet Formalism
+
+Author: C. Staufenbiel, 2022
+
+inspirations taken from the [floquet notebook](https://github.com/qutip/qutip-notebooks/blob/master/examples/floquet-dynamics.ipynb) by P.D. Nation and J.R. Johannson,
+
+and the [qutip documentation](https://qutip.org/docs/latest/guide/dynamics/dynamics-floquet.html).
+
+### Introduction
+In the [floquet_solver notebook](011_floquet_solver.md) we introduced the two functions to solve the Schrödinger and Master equation using the Floquet formalism. In this notebook, we will focus on the internal functions of these solvers, that implement the Floquet formalism in QuTiP. Here, we will focus on the `floquet modes` and the `quasienergies`.
+
+More information on the implementation of the Floquet Formalism in QuTiP can be found in the [documentation](https://qutip.org/docs/latest/guide/dynamics/dynamics-floquet.html).
+
+### Imports 
+
+```python
+import matplotlib.pyplot as plt
+import numpy as np
+from qutip import (about, expect, floquet_markov_mesolve,
+                   floquet_master_equation_rates,
+                   floquet_master_equation_tensor, floquet_modes,
+                   floquet_modes_t, floquet_modes_t_lookup,
+                   floquet_modes_table, floquet_state_decomposition,
+                   floquet_wavefunction, floquet_wavefunction_t, fmmesolve,
+                   ket2dm, num, plot_wigner, rand_ket, sesolve, sigmax, sigmaz)
+```
+
+### System setup
+For consistency with the documentation we consider the driven system with the following Hamiltonian: 
+
+$$ H = - \frac{\Delta}{2} \sigma_x - \frac{\epsilon_0}{2} \sigma_z + \frac{A}{2} \sigma_x sin(\omega t) $$
+
+```python
+# Constants
+delta = 0.2 * 2 * np.pi
+eps0 = 1 * 2 * np.pi
+A = 2.5 * 2 * np.pi
+omega = 1.0 * 2 * np.pi
+T = 2 * np.pi / omega
+
+# Hamiltonian
+H = [
+    -delta / 2.0 * sigmax() - eps0 / 2.0 * sigmaz(),
+    [A / 2 * sigmax(), "sin({w}*t)".format(w=omega)],
+]
+```
+
+### Floquet modes and quasienergies
+For periodic Hamiltonians the solution to the Schrödinger equation can be represented by the Floquet modes $\phi_\alpha(t)$ and the quasienergies $\epsilon_\alpha$. We can obtain these for the initial time $t=0$ by using the function `floquet_modes()`. We can display for example the first floquet mode at $t=0$ using a Wigner distribution.
+
+```python
+f_modes0, f_energies = floquet_modes(H, T)
+plot_wigner(f_modes0[0]);
```

Comment:
this is an interesting use of plot_wigner!  can we discern anything from this?

Reply:
I just wanted to show that the floquet mode is a typical quantum state. I guess for a more complicated Hamiltonian the Wigner plot would be more interesting.